### ISSUE TITLE
When using multiprocessing with unix socket, http response is sent only after the process finish

### ISSUE THREAD

**Issue by autra** [unbit/uwsgi#1711]
## Description

When using uwsgi in a flask application that uses multiprocessing, the http response is only sent when the process started by multiprocessing finishes. 

This does not happen when:
- using a http port with uwsgi
- starting the application directly with the `flask` command-line

## Steps to reproduce

- clone https://github.com/autra/uwsgi_bug_test_case
It contains this flask application:
```python
import flask

app = flask.Flask(__name__)

def foo():
    print('executing foo')
    result = 0
    for i in range(100000000):
        result = i*i
    print('end executing foo', result)
    return result

@app.route('/test/')
def test():
    import multiprocessing
    process = multiprocessing.Process(target=foo)
    process.start()
    print('returning ok')
    return 'ok'

```
- follow instruction on README (create a virtualenv and install stuff in it, start the server...)

**expected results**

Whether with `uwsgi.yml`, `uwsgi_bug.yml` or with `FLASK_APP=app/app.py flask run`, visiting the `/test/` route in a browser should immediately display ok, and some second later, the console should display `end executing foo`.

**actual results**
When using uwsgi_bug.yml, the http response is not sent to the browser until `end executing foo` appears.

On a broader context, I'm actually trying to integrate https://github.com/geopython/pywps/ in my own flask integration. I therefore cannot easily stop using multiprocessing.

Please let me know if you need more information.

**Comment by xrmx** [comment]
From a very quick search this does not use multiprocessing so are you sure you need it? https://github.com/geopython/pywps-flask

**Comment by elemoine** [comment]
The underlying lib does: https://github.com/geopython/pywps/blob/48d0e6b5bc5c3f31b3d5eb56341d4a66558ccd51/pywps/processing/basic.py#L32.

**Comment by xrmx** [comment]
Yeah but that code is not something you'll run from a worker, looks more like a service on its own

**Comment by autra** [comment]
@xrmx thanks for your answer. Does uwsgi intent to support the use of `multiprocessing` like it is done in the snippet above? PyWPS is indeed our broader context, but this issue could be valid independently of this fact.

**Comment by xrmx** [comment]
Nope, you should probably have a pool of pywps workers and have your web app talk to them via some kind of ipc. Or use a task queue like celery.

**Comment by dlparker** [comment]
I ran into this as well, and found I can work around it by using a thread to start the process. Note the commented lines with the workaround. Not sure yet if this produces a reliable long running process since I haven't tested thoroughly, but it solves the initial problem.  In this case it is that the request that starts the process hangs.

My question is, should I have seen this restriction in the documentation somewhere? 

```
import json
import time
import threading
import multiprocessing as mp
from flask import Flask
init_done = False

app = Flask(__name__)

def worker():
    while(True):
        time.sleep(0.1)

def cmd_init_thread():
    global init_done
    if init_done:
        return
    p = mp.Process(target=worker)
    p.start()
    print('init done')
    init_done = True

@app.route('/ping', methods=['GET'])
def ping():
    global init_done
    if not init_done:
        # comment this line and uncomment the thread ops and it works                                                                    
        cmd_init_thread()
        #thread = threading.Thread(target=cmd_init_thread, daemon=True)                                                                  
        #thread.start()                                                                                                                  
        print('calling init from ping')
    return json.dumps({"pong": True})

if __name__ == "__main__":
    app.run()
```
